### ISSUE TITLE
Aligning HTML spec with ARIA's style of DOMString? reflection

### ISSUE THREAD

**Issue by rahimabdi** [w3c/aria#2414]
# Describe your concern

Many non-enumerated ARIA attributes currently string-reflect as `DOMString?` (nullable DOMString) such as `aria-label`. This is desirable to explicitly distinguish when an attribute is missing which returns `null` for both the content attribute and IDL attribute.  To see the IDL attribute type for all ARIA content attributes, please see [section 10.1 ARIA Mixin interface](https://w3c.github.io/aria/#ARIAMixin). 

However, per HTML reflection, `DOMString?` requires that the content attribute be enumerated. From [HTML 2.6.1 Reflecting content attributes in IDL attributes](https://html.spec.whatwg.org/#reflecting-content-attributes-in-idl-attributes), below is the processing model for the `DOMString?` getter:

> 1. Let element be the result of running [this](https://webidl.spec.whatwg.org/#this)'s [get the element](https://html.spec.whatwg.org/#get-the-element).
> 
> 2. Let contentAttributeValue be the result of running [this](https://webidl.spec.whatwg.org/#this)'s [get the content attribute](https://html.spec.whatwg.org/#get-the-content-attribute).
> 
> 3. Let attributeDefinition be the attribute definition of element's content attribute whose namespace is null and local name is the [reflected content attribute name](https://html.spec.whatwg.org/#reflected-content-attribute-name).
> 
> 4. [Assert](https://infra.spec.whatwg.org/#assert): attributeDefinition indicates it is an [enumerated attribute](https://html.spec.whatwg.org/#enumerated-attribute).

Many ARIA IDL attributes that reflect as `DOMString?` do not conform with HTML `DOMString?` reflection due to the assertion failure in step 4 (i.e., attribute is enumerated). 

After discussion with @cookiecrook @annevk, one path forward that @annevk is OK with is loosening HTML's reflection model to allow ARIA's IDL use case of string reflection for `DOMString?` and essentially removing the enumerated attribute assertion check. The primary benefits of this change is closer alignment with the behavior of HTML attributes and HTML's reflection model. If there are no objections from the ARIA WG, I will submit a PR to update the HTML spec's `DOMString?` reflection model as part of this existing HTML issue: https://github.com/whatwg/html/issues/10037.

# Link to the version of the specification or documentation you were looking at at.

Link to documentation: https://w3c.github.io/aria/

Does the issue exists in [the editors draft](https://w3c.github.io/aria/) (the editors draft is the most recent draft of the specification)?
Yes

**Comment by rahimabdi** [comment]
This was discussed in the Jan 30 2025 ARIA WG meeting today and there was consensus to move forward with the HTML spec update.

**Comment by smockle** [comment]
@rahimabdi The IDL project was restored, if you want to use it to track this and related issues: https://github.com/orgs/w3c/projects/176

**Comment by rahimabdi** [comment]
@smockle Thank you!

**Comment by rahimabdi** [comment]
This issue can be closed as the HTML PR has been merged: https://github.com/whatwg/html/pull/10974. The corresponding HTML issue can be found here: https://github.com/whatwg/html/issues/10037.

The new `DOMString?` getter can be found in [HTML 2.6.1 Reflecting content attributes in IDL Attributes](https://html.spec.whatwg.org/#reflecting-content-attributes-in-idl-attributes).

Thank you @cookiecrook @annevk for your support!
